src/widgets/editor.rs

```diff
@@ -118,9 +120,77 @@ impl Future for Editor {
 
 impl Editor {
     /// Handle keyboard and mouse events
-    pub fn handle_input(&mut self, event: TermionEvent) {
-        if let Some(view) = self.views.get_mut(&self.current_view) {
-            view.handle_input(event)
+    pub fn handle_input(&mut self, event: Event) {
+        // We have to remove and insert again, to beat the borrow-checker
+        match event {
+            Event::Key(key) => {
+                match self.keybindings.keymap.get(&event).cloned() {
+                    Some(cmd) => self.handle_command(cmd),
+                    None => { 
+                        if let Some(view) = self.views.get_mut(&self.current_view) {
+                            match key {
+                                Key::Char(c) => match c {
+                                    '\n' => view.insert_newline(),
+                                    '\t' => view.insert_tab(),
+                                    _ => view.insert(c),
+                                },
+                                k => error!("un-handled key {:?}", k)
+                            }
+                        }
+                    },
+                }
+            },
+            Event::Mouse(mouse_event) => self.views.get_mut(&self.current_view).unwrap().handle_mouse_event(mouse_event),
+            ev => error!("un-handled event {:?}", ev),
+        }
+    }
+
+    pub fn handle_command(&mut self, cmd: Command) {
+        match cmd {
+            Command::Cancel => {/* Handled by TUI */}
```

Comment:
I think this can't even be reached right? Then in the default case we could have:

```rust
match cmd {
    ...
    _ => unreachable!()
}
```

Reply:
This code is again different on HEAD. 
The only place where I do not have a default case is in the `Client`, because I *want* a warning when I forgot to add a case.
The `/* Handled by ... */` comments are just there as a sort of documentation